api/src/main/java/edu/cornell/mannlib/vitro/webapp/audit/controller/AuditController.java

```diff
@@ -0,0 +1,334 @@
+                body.put("username", acc.getFirstName());
+            }
+        } else if (StringUtils.isNotEmpty(acc.getEmailAddress())) {
+            body.put("username", acc.getEmailAddress());
+        } else {
+            body.put("username", "");
+        }
+
+        // Pass the helper methods to Freemaker
+        body.put("listAddedStatements", new ListAddedStatementsMethod());
+        body.put("listRemovedStatements", new ListRemovedStatementsMethod());
+
+        // Return the default template and parameters
+        return new TemplateResponseValues(TEMPLATE_DEFAULT, body);
+    }
+
+    private boolean isAdmin(UserAccount acc) {
+        if (acc.isRootUser()) {
+            return true;
+        }
+        Set<String> roles = acc.getPermissionSetUris();
+        return (roles.contains(PermissionSets.URI_DBA));
+    }
+
+    private void setUserData(List<AuditChangeSet> list, UserAccountsDao uad) {
+        for (AuditChangeSet acs : list) {
+            UserAccount account = uad.getUserAccountByUri(acs.getUserId());
+            if (account == null) {
+                continue;
+            }
+            acs.setUserFirstName(account.getFirstName());
+            acs.setUserLastName(account.getLastName());
+            acs.setUserEmail(account.getEmailAddress());
+        }
+    }
+
+    /**
+     * Get the number of entries to show per page, or a default value of 10
+     *
+     * @param vreq
+     * @return
+     */
+    private int getLimit(VitroRequest vreq) {
+        int limit = 0;
+        try {
+            limit = Integer.parseInt(vreq.getParameter(PARAM_LIMIT));
+        } catch (Throwable e) {
+            log.debug(e, e);
+            limit = 10;
+        }
+        return limit;
+    }
+
+    /**
+     * Get the page offset, or a default of 0
+     *
+     * @param vreq
+     * @return
+     */
+    private int getOffset(VitroRequest vreq) {
+        int offset = 0;
+        try {
+            offset = Integer.parseInt(vreq.getParameter(PARAM_OFFSET));
+        } catch (Throwable e) {
+            log.debug(e, e);
+            offset = 0;
+        }
+        return offset;
+    }
+
+    /**
+     * Get start date
+     *
+     * @param vreq
+     * @return
+     */
+    private Date getStartDate(VitroRequest vreq) {
+        String start = vreq.getParameter(PARAM_START_DATE);
+        try {
+            if (!StringUtils.isBlank(start)) {
+                Date startDate = sdf.parse(start);
+                return startDate;
+            }
+        } catch (Exception e) {
+            log.error(e, e);
+        }
+        return DateUtils.addDays(new Date(), -7);
+    }
+
+    /**
+     * Get end date
+     *
+     * @param vreq
+     * @return
+     */
+    private Date getEndDate(VitroRequest vreq) {
+        String end = vreq.getParameter(PARAM_END_DATE);
+        try {
+            if (!StringUtils.isBlank(end)) {
+                Date endDate = sdf.parse(end);
+                return endDate;
```

Comment:
Why not just return the response to the parse method call instead of temporarily allocating memory for garbage cleanup?

Reply:
fixed